### PR TITLE
fix: remove compass-preferences from connection-form completely COMPASS-8098

### PR DESCRIPTION
**Pull request by paula-stacho**
## Description

Removing last reference to compass-preferences from connection-form, because it's causing some issues in the VSCode integration.

### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context

- [ ] Bugfix
- [ ] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/connection-form/src/hooks/use-connection-color.ts [hunk 1]
```diff
@@ -188,22 +171,10 @@ export function useConnectionColor(): {
...
-      return CONNECTION_COLOR_CODES;
-    }
-  };
+  const connectionColorCodes = () => CONNECTION_COLOR_CODES.slice(0, 9);
 
   return {
-    connectionColorToHex: isMultiConnectionEnabled
-      ? newColorCodeToHex
-      : colorCodeToHex,
+    connectionColorToHex: newColorCodeToHex,
```
Comment (Anemy) [inline packages/connection-form/src/hooks/use-connection-color.ts:177]: [nit] Does it make sense to drop the `new` now?

File: packages/connection-form/src/hooks/use-connection-color.ts [hunk 2]
```diff
@@ -161,22 +160,6 @@ export function useConnectionColor(): {
     [isDarkMode]
   );
 
-  const colorCodeToHex = useCallback(
-    (colorCode: string | undefined): string | undefined => {
-      if (!colorCode) {
-        return;
-      }
-
-      const migratedColor = legacyColorsToColorCode(colorCode);
```
Comment (Anemy) [inline packages/connection-form/src/hooks/use-connection-color.ts:170]: We only have one other use of `legacyColorsToColorCode`, and it's in the color picker. Maybe we can remove `legacyColorsToColorCode` and `LEGACY_COLORS_TO_COLOR_CODE_MAP` now? Someone would be using a color from ~3 years ago, and we'd only slightly be helping them select a color inside of the color picker UI.

**Review by Anemy — APPROVED** [review]
lgtm! One suggestion on some old code we can remove, not a blocker, feel free to merge as is

File: packages/connection-form/src/hooks/use-connection-color.ts (hunk 2 above)
Comment (paula-stacho) [reply packages/connection-form/src/hooks/use-connection-color.ts:170]: that must be part of the legacy bits. I'll try to remove it + the parts that use it, hopefully it's not too deep of a rabbit hole 🤞

**Pull request merged**